cache/lru/cache.go

```diff
@@ -55,15 +77,22 @@ func (c *Cache[K, V]) Get(key K) (V, bool) {
 func (c *Cache[K, _]) Evict(key K) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-
+	if value, ok := c.elements.Get(key); ok {
+		c.onEvict(key, value)
+	}
 	c.elements.Delete(key)
 }
 
 func (c *Cache[_, _]) Flush() {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	c.elements.Clear()
+	// Call onEvict for each element before clearing
+	iter := c.elements.NewIterator()
+	for iter.Next() {
+		c.onEvict(iter.Key(), iter.Value())
+		c.elements.Delete(iter.Key())
```

Comment:
Since we need to call the `onEvict` callback... it feels like `Flush` is now becoming an implementation of `Evict` over the entire `Cache`. Would it make sense to refactor the locked part of `Evict` into something like `evict` so we can just call `evict` over each element in this loop so we can avoid re-implementing `Evict` here?